server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/UserLdapGroup.java

```diff
@@ -0,0 +1,117 @@
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.server.GenericOperationResult;
+import com.walmartlabs.concord.server.OperationResult;
+import com.walmartlabs.concord.server.sdk.ConcordApplicationException;
+import com.walmartlabs.concord.server.security.Roles;
+import com.walmartlabs.concord.server.user.CreateUserRequest;
+import com.walmartlabs.concord.server.user.UserInfoProvider;
+import com.walmartlabs.concord.server.user.UserManager;
+import com.walmartlabs.concord.server.user.UserType;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.Authorization;
+import org.apache.shiro.authz.UnauthorizedException;
+import org.sonatype.siesta.Resource;
+import org.sonatype.siesta.Validate;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.Set;
+import java.util.UUID;
+
+
+@Named
+@Singleton
+@Api(value = "UserLdapGroup", authorizations = {@Authorization("api_key"), @Authorization("session_key"), @Authorization("ldap")})
+@Path("/api/v1/userldapgroup")
+public class UserLdapGroup implements Resource {
+
+    private final UserManager userManager;
+    private final LdapUserInfoProvider ldapUserInfoProvider;
+    private final LdapGroupDao ldapGroupsDao;
+    private final LdapManager ldapManager;
+
+    @Inject
+    public UserLdapGroup(UserManager userManager, LdapUserInfoProvider ldapUserInfoProvider, LdapGroupDao ldapGroupsDao, LdapManager ldapManager) {
+        this.userManager = userManager;
+        this.ldapUserInfoProvider = ldapUserInfoProvider;
+        this.ldapGroupsDao = ldapGroupsDao;
+        this.ldapManager = ldapManager;
+    }
+
+    /**
+     * Sync Ldap groups for a ldap user
+     *
+     * @param req user's data
+     * @return GenericOperationResult result
+     */
+    @POST
+    @ApiOperation("Sync ldap groups for a user")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/sync")
+    @Validate
+    public GenericOperationResult sync(@ApiParam @Valid CreateUserRequest req) {
```

Comment:
I think it would be better to use a new type here instead of re-using `CreateUserRequest`. There are fields in the `CreateUserRequest` class that are not relevant for this method.

Reply:
sounds good. updated